lib/Instance_spec.lua

```diff
@@ -146,4 +146,40 @@ describe("Instance", function()
 			assert.equal(tostring(instance), "foo")
 		end)
 	end)
+
+	describe("Changed", function()
+		it("should fire Changed", function()
+			local instance = Instance.new("Folder")
+
+			local changedSpy = spy.new(function() end)
+			instance.Changed:Connect(changedSpy)
+
+			instance.Name = "NameChange"
+			assert.spy(changedSpy).was.called_with("Name")
+		end)
+	end)
+
+	describe("GetPropertyChangedSignal", function()
+		local instance = Instance.new("Folder")
+
+		it("should fire property signals for the right property", function()
+			local spy = spy.new(function() end)
+			instance:GetPropertyChangedSignal("Name"):Connect(spy)
+			instance.Name = "NameChange"
+			assert.spy(spy).was.called()
+		end)
+
+		it("should not fire property signals for the incorrect property", function()
+			local spy = spy.new(function() end)
+			instance:GetPropertyChangedSignal("Parent"):Connect(spy)
+			instance.Name = "NameChange2"
+			assert.spy(spy).was_not_called()
+		end)
+
+		it("should error when given an invalid property name", function()
+			assert.has.errors(function()
+				instance:GetPropertyChangedSignal("CanDestroyTheWorld"):Connect(function() end)
+			end)
+		end)
```

Comment:
These tests should all create their own instances, otherwise you'll get a bunch of signal connections that last through each test!